hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/container/common/helpers/BlockData.java

```diff
@@ -92,15 +94,13 @@ public void setBlockCommitSequenceId(long blockCommitSequenceId) {
    *
    * @param data - Protobuf data.
    * @return - BlockData
-   * @throws IOException
    */
-  public static BlockData getFromProtoBuf(ContainerProtos.BlockData data) throws
-      IOException {
+  public static BlockData getFromProtoBuf(ContainerProtos.BlockData data) throws CodecException {
     BlockData blockData = new BlockData(
         BlockID.getFromProtobuf(data.getBlockID()));
     for (int x = 0; x < data.getMetadataCount(); x++) {
-      blockData.addMetadata(data.getMetadata(x).getKey(),
-          data.getMetadata(x).getValue());
+      final ContainerProtos.KeyValue meta = data.getMetadata(x);
+      blockData.addMetadata(meta.getKey(), meta.getValue(), CodecException::new);
     }
     blockData.setChunks(data.getChunksList());
     if (data.hasSize()) {
```

Comment:
Should we throw `CodecException` on size mismatch?

https://github.com/apache/ozone/blob/a354d00bb01f195592b74a9b359170947d67399c/hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/container/common/helpers/BlockData.java#L106-L107